keras_cv/layers/regularization/squeeze_excite.py

```diff
@@ -25,12 +24,14 @@ class SqueezeAndExciteBlock2D(layers.Layer):
 
     Args:
         filters: Number of input and output filters. The number of input and
-                output filters is same.
+            output filters is same.
         ratio: Ratio for bottleneck filters. Number of bottleneck filters =
-                filters * ratio. Defaults to 0.25.
-        squeeze_activation: (Optional) String or function denoting activation to
+            filters * ratio. Defaults to 0.25.
+        squeeze_activation: (Optional) String, function or tf.keras.activations.*
+            instance denoting activation to
             be applied after squeeze convolution. Defaults to `relu`.
-        excite_activation: (Optional) String or function denoting activation to
+        excite_activation: (Optional)String, function or tf.keras.activations.*
```

Comment:
replace function with "callable" (or tf.keras.Layer), replace tf.keras.activations.* with "tf.keras.activations.Activation" -> we have a base class for these.

Also you forgot a space after optional :)

Reply:
@LukeWood even with the change in serialization_test, it errors out for activation functions.

Reply:
This one is a bit weird. Here's the test case:
```py
        (
            "SqueezeAndExciteBlock2D",
            regularization.SqueezeAndExciteBlock2D,
            {
                "filters": 16,
                "ratio": 0.25,
                "squeeze_activation": tf.keras.layers.ReLU(),
                "excite_activation": tf.keras.activations.relu,
            },
        ),

```
Error trace: https://codeshare.io/3AMwqv

I have inserted the print statements as follows:
```py
def custom_compare(obj1, obj2):
    print("#" * 20, "custom_compare")

    if isinstance(obj1, (core.FactorSampler, tf.keras.layers.Layer)):
        print(obj1.get_config())
        print(obj2.get_config())
        return config_equals(obj1.get_config(), obj2.get_config())
    else:
        return obj1 == obj2


def config_equals(config1, config2):
    print("#"*20, "config_equals")
    print(config1)
    print(config2)
    for key in list(config1.keys()) + list(config2.keys()):
        v1, v2 = config1[key], config2[key]
        if not custom_compare(v1, v2):
            return False
    return True

```

Reply:
Note that the function is getting "auto-serialized":

```
original: {'name': 'squeeze_and_excite_block2d', 'trainable': True, 'dtype': 'float32', 'filters': 16, 'ratio': 0.25, 'squeeze_activation': <keras.layers.activation.relu.ReLU object at 0x0000025EC9F39208>, 'excite_activation': <function relu at 0x0000025EF179B318>}
restored: {'name': 'squeeze_and_excite_block2d', 'trainable': True, 'dtype': 'float32', 'filters': 16, 'ratio': 0.25, 'squeeze_activation': <keras.layers.activation.relu.ReLU object at 0x0000025EF1D86088>, 'excite_activation': 'relu'}
```

Reply:
The paste is empty.

Reply:
This code is still working: it is serializing and deserializing a relu activation.  the code should not be changed to work around the test case.  The test case should be changed to work around the code 👍

Reply:
Could you try this one: https://www.codepile.net/pile/nP5Nnyl4